src/emit.rs

```diff
@@ -59,12 +59,15 @@ pub struct EmittedSourceText {
   pub source_map: Option<String>,
 }
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, deno_error::JsError)]
 pub enum EmitError {
+  #[class(inherit)]
   #[error(transparent)]
-  SwcEmit(std::io::Error),
+  SwcEmit(#[inherit] std::io::Error),
+  #[class(TYPE)]
   #[error(transparent)]
   SourceMap(sourcemap::Error),
```

Comment:
not sure if we want to have this be a TypeError or handle all the `sourcemap::Error` variants